Documentation/ceph-object-bucket-notifications.md

```diff
@@ -0,0 +1,171 @@
+See the Ceph documentation for detailed information: https://docs.ceph.com/en/latest/radosgw/notifications/.
+- a `CephBucketTopic` is custom resource which represents a bucket notification topic and is described by a CRD shown below. A bucket notification topic represents an endpoint (or a "topic" inside this endpoint) to which bucket notifications could be sent.
+
+## Notifications
+A CephBucketNotification defines what bucket actions trigger the notification and which topic to send notifications to. A CephBucketNotification may also define a filter, based on the object's name and other object attributes. Notifications can be associated with buckets created via ObjectBucketClaims by adding labels to an ObjectBucketClaim with the following format:
+```yaml
+bucket-notification-<notification name>: <notification name>
+```
+The CephBucketTopic, CephBucketNotification and ObjectBucketClaim must all belong to the same namespace.
+If a bucket was created manually (not via an ObjectBucketClaim), notifications on this bucket should also be created manually. However, topics in these notifications may reference topics that were created via CephBucketTopic resources.
+
+## Topics
+A CephBucketTopic represents an endpoint (of types: Kafka, AMQP0.9.1 or HTTP), or a specific resource inside this endpoint (e.g a Kafka or an AMQP topic, or a specific URI in an HTTP server). The CephBucketTopic also holds any additional info needed for a CephObjectStore's RADOS Gateways (RGW) to connect to the endpoint. Topics don't belong to a specific bucket or notification. Notifications from multiple buckets may be sent to the same topic, and one bucket (via multiple CephBucketNotifications) may send notifications to multiple topics.
+
+## Notification Reliability and Delivery
+Notifications may be sent synchronously, as part of the operation that triggered them. In this mode, the operation is acknowledged only after the notification is sent to the topic’s configured endpoint, which means that the round trip time of the notification is added to the latency of the operation itself.
+The original triggering operation will still be considered as successful even if the notification fail with an error, cannot be delivered or times out.
+
+Notifications may also be sent asynchronously. They will be committed into persistent storage and then asynchronously sent to the topic’s configured endpoint. In this case, the only latency added to the original operation is of committing the notification to persistent storage.
+If the notification fail with an error, cannot be delivered or times out, it will be retried until successfully acknowledged.
+
+## Sample
+
+### CephBucketTopic Custom Resource
+```yaml
+apiVersion: objectbucket.io/v1alpha1
+kind: CephBucketTopic
+metadata:
+  name: my-topic [1]
+  namespace: my-app-space [2]
+spec:
+   # use one of the following endpoints
+  endpoint: http://my-notification-endpoint:8080 [3]
```

Comment:
Is there a tab/space issue happening here? I usually have the VSCode option "Render Whitespace" set to "boundary".